streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -342,7 +342,19 @@ public void handleAssignment(final Map<TaskId, Set<TopicPartition>> activeTasks,
 
         maybeThrowTaskExceptions(taskCloseExceptions);
 
-        createNewTasks(activeTasksToCreate, standbyTasksToCreate);
+        rebalanceInProgress = true;
+        final Collection<Task> newActiveTasks = createNewTasks(activeTasksToCreate, standbyTasksToCreate);
```

Comment:
Seems we only the count (or a boolean) if empty or not, but not the full collection. Can we simplify this?

Reply:
The call for `createNewTasks` was already there in `handleAssignment` to create new tasks. I just modified its return type to check if it's empty or not to trigger the tasks commit and offset commit.